### PR TITLE
[desktop] reduce API surface of preload script after it has run

### LINKED ISSUE #1620: use ipcRenderer.invoke / ipcMain.handle instead of remote
as of 7.0.0, there are new recommended functions for ipc. see
https://electronjs.org/blog/electron-7-0

### PR DESCRIPTION
**Pull request by ganthern**
also close #1620 since the new ipc api is essentially what we're doing, only not fully bidirectional.

### CODE REVIEW

File: src/desktop/preload.js [hunk 1]
```diff
@@ -1,27 +1,43 @@
 // @flow
-import {ipcRenderer, remote} from 'electron'
 
 /**
  * preload scripts can only load modules that have previously been loaded
  * in the main thread.
  */
+let remote = require('electron').remote
```
Comment (charlag) [inline src/desktop/preload.js:7]: Do you need to do `require('electron')` now? You also do it multiple times. I would expect that it compiles to the same thing but maybe I'm wrong.

**Comment by ganthern** [comment]
> In the #1620 you say that we should use `invoke()` instead of remote but we still use `send()`, can't we use `invoke()` now if we do bookkeeping anyway?

`invoke()` is essentially a wrapper around `send()` that works similarly to what we're doing with the nativeApp/nativeCommands (so it does the bookkeeping), but doesn't pass messages in the main->renderer direction.

**Comment by charlag** [comment]
I looked through the code and wanted to try it out but noticed that it's not rebased on the master (after Flow was updated). Could you please do that? Currently PR commit (4aecdce9) is based on 07b261fc (28th of April) and Flow update was in 14f01b3e (which was on 30th or so). I could do it as well but if I think it might be easier for you.

**Comment by ganthern** [comment]
done!

**Pull request merged**